addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -225,17 +229,21 @@ def _get_tax_category_list(self, invoice, taxes):
     # -------------------------------------------------------------------------
 
     def _check_required_fields(self, record, field_names, custom_warning_message=""):
-        """
-        This function check that a field exists on a record or dictionaries
-        returns a generic error message if it's not the case or a custom one if specified
+        """Check if at least one of the field_names are set on the record/dict
+
+        :param record: either a recordSet or a dict
+        :param field_names: The field name or list of field name that has to
+                            be checked. If a list is provided, check that at
+                            least one of them is set.
+        :return: an Error message or None
         """
         if not record:
             return custom_warning_message or _("The element %s is required on %s.", record, ', '.join(field_names))
 
-        if not isinstance(field_names, list):
-            field_names = [field_names]
+        if not isinstance(field_names, (list, tuple)):
+            field_names = (field_names,)
 
-        has_values = any(record[field_name] for field_name in field_names)
+        has_values = any((field_name in record and record[field_name]) for field_name in field_names)
```

Comment:
Look like `.get()` should do the trick here

Reply:
No for 2 reasons:
- the method has to work with a record (and records don't have get method)
- if there are valid fields with falsy value, that won't work

Reply:
Damn, don't read the docstring.. :grimacing: